### PR TITLE
add views specs and fix n+1 problems

### PR DESCRIPTION
**Pull request by diegoh40**
- Write integration tests with Capybara gem.
- Secure app from n+1 problems.

### CODE REVIEW

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has a readme missing the key features and future features section.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: Gemfile [hunk 1]
```diff
@@ -54,7 +52,9 @@ gem 'bootsnap', require: false
 
 group :development, :test do
   # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
+  gem 'database_cleaner'
   gem 'debug', platforms: %i[mri mingw x64_mingw]
+  gem 'rails-controller-testing'
   gem 'rspec-rails', '~> 5.0.0'
 end
```
Comment (mirouhml) [inline Gemfile:59]: - [ ] Please know that you should add `capybara` and not in the test group, kindly adjust that.

File: spec/views/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    end
+
+    it 'should return username' do
+      expect(page).to have_content('Diego')
+    end
+
+    it 'should return user posts' do
+      expect(page).to have_content('Number of Posts: 0')
+    end
+  end
```
Comment (mirouhml) [inline spec/views/post_index_spec.rb:37]: - [ ] Please know that you should implement all the test mentioned in the requirements, these are the ones missing:
  * Tests that asserts you can see a post's title.
  * Tests that asserts you can see some of the post's body.
  * Tests that asserts you can see the first comments on a post.
  * Tests that asserts you can see how many comments a post has.
  * Tests that asserts you can see how many likes a post has.
  * Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
  * Tests that redirects to the post's show page when you click on a post.
- [ ] Also note that the `context` is wrong here you should write something meaningful that shows what you're testing, kindly adjust that.
- [ ] You've also repeated the first 3 tests twice, kindly remove the redundant ones.

**Review by mirouhml — CHANGES_REQUESTED** [review]
# REQUIRED CHANGES 🔴

Hi @diegoh40 @robinbenitezmora, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Great job setting up the correct linters ✔️
- Highly organized and clean code ✔️ 👨🏻‍💻
- Professional documentation ✔️
- Correct workflow ✔️
- Meaningful commit messages ✔️

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**